### ISSUE TITLE
File postions in boss/logex/fmpsee limited to 31 bits

### ISSUE THREAD

**Issue by wehimwich** [nvi-inc/fs#115]
The variables that handle file positions in _fmpsee/_ library routines `fmpsetpos()` and `fmpposition()` are `int` but should be `long` instead. They were changed in the bulk _unlongify_ of c9cbc93b94a71944353704d33182269bb1706d1d and never changed back.

Fortunately, this is benign as long as the files are less than 2<sup>31</sup> bytes long. Since _boss_ only uses file positions for _.snp_ and _.prc_ files, this is unlikely to create a problem, even on 64-bit systems. It seems likely that this breaks at least some parts of _logex_; It also seems likely that no one uses it anymore (let us know if you do). These reasons, and the amount of work to fix the calling FORTRAN, probably made this clean-up a low priority task. In fact, switching to `int` variables could be viewed as a win in the short-term. That may have been the actual decision at the time, but if it was (things were busy), I don't remember it. Anyway, the same limitation existed in the previous 32-bit only FS versions.

In _boss_, the FORTRAN generally has the affected variables declared `integer*4`. To extend the range for 64-bit systems, these variables will probably need to be changed into two element arrays. Some of these values are also stored in other `integer` arrays. Some adjustment will be needed for that. It seem unlikely any of these variables (or array elements) are used in arithmetic but that should be checked for carefully as well.

In _logex_, at least some of the involved variables are declared `integer*2`, so it seems likely that some commands, probably  at least `plot`, won't work, even on 32-bit systems. It seems unlikely anyone is still using _logex_, but it may not take much work to clean this up.

In the same commit,  the line number variables in the `fmpsetline()` routine were also changed to `int`. Whether _.snp_ files with 2<sup>31</sup> or more lines need to be supported is a different question. If so, it  would have ramifications in the line number handling in _boss_ as well. This appears to be an issue for _logex_ for processing _.log_ files.

Pieces needed to support larger _.snp_ and _.prc_ files:

- [ ] Fix `fmpsetpos()` and `fmpposition()`
- [ ] Expand size of affected FORTRAN variables
- [ ] Accommodate larger variables when stored in FORTRAN arrays
- [ ] Check for arithmetic with  FORTRAN file positions variables/arrays

Line numbers in _.skd_ files:

- [ ] Consider/fix how many lines should be allowed in _.snp_ files (`fmpsetline()` and more generally in _boss_).

_logex_:

- [ ] Consider fixing _logex_

**Comment by dehorsley** [comment]
Good catch. I think some of these incorrect `int`s could be caught by the compiler  (with `-Wconversion` I think), but I seem to remember there was too many false positives.
